### PR TITLE
[API] Update API tests with cache

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

This is a followup [to the this comment on](https://github.com/Recidiviz/pulse-dashboard/pull/671#discussion_r548197104) PR #671 to include more of the API code in the tests. 

Instead of mocking the redis client, I opted for checking the `NODE_ENV` when creating the cache, and using a memory store for the test environment instead of redis. This means we're not including the packages `ioredis` and `cache-manager-ioredis` in the tests, but it is still testing the functionality of `cache-manager`. I opted for this instead of running a redis server in the test environment, but I could be convinced that we want to do that instead. Let me know what you all think!

The API test is set up to expect `fetchMetrics` to be called only when the data is not in the cache. I didn't mock out objectStorage for these tests since `fetchMetrics`, `fetchMetricsFromGCS`, and `fetchMetricsFromLocal` already have unit tests. 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Part of #652 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: server/core/redisCache.js [hunk 1]
```diff
@@ -46,32 +46,34 @@ const redisInstance = new Redis({
...
+  store: testEnv ? "memory" : redisStore,
   refreshThreshold: REDIS_CACHE_REFRESH_THRESHOLD,
   redisInstance,
 });
 
-const redisClient = redisCache.store.getClient();
-redisClient.on("error", (error) => {
-  console.error("ERR:REDIS:", error);
-});
+if (process.env.NODE_ENV === "production") {
```
[the remark this answers is not in the document]
Comment (daschi) [reply server/core/redisCache.js:57]: Oh it should, I'll change this to check that we're not in the test environment instead!

**Comment by macfarlandian** [comment]
Sorry I needed to ponder this for a bit ... I think it's reasonable to just always use a memory cache in tests, since there isn't any difference in the cache logic itself. (The redis integration code is ) I think it would be preferable if the mechanism for doing this was a bit less obscure, though ... e.g., if the endpoint function used a flag to determine whether it should use `memoryCache` or `redisCache`, rather than the `redisCache` secretly not actually being a Redis cache? I don't feel super strongly about this though so I am curious what others think

**Review by jovergaag — COMMENTED** [review]
> I think it would be preferable if the mechanism for doing this was a bit less obscure, though ... e.g., if the endpoint function used a flag to determine whether it should use `memoryCache` or `redisCache`, rather than the `redisCache` secretly not actually being a Redis cache

I agree with this. It is a little misleading that the redisCache is in fact memory in the tests. 

If the endpoint function did use a flag of some sort, even possibly based on the requested file, we could make the endpoints be much more flexible and quickly and easily move certain files to redis if the instance's memory starts to fill up rather than moving an entire endpoint at a time.

**Comment by daschi** [comment]
@macfarlandian and @jovergaag I just pushed up a commit to change it around a bit and combine the redis and memory caches in one file and export a helper `cacheResponse` that will have access to the logic for which cache store to use. Let me know if this is in line what you were thinking!

**Review by jovergaag — APPROVED** [review]
I think that's a great change. Much more flexible and clear!

**Review by macfarlandian — APPROVED** [review]

**Pull request merged**